### ISSUE TITLE
Reading BAM files

### ISSUE THREAD

**Issue by mikael-s** [GATB/gatb-core#40]
Hi!

Is there any possibility to read a BAM file as a bank of sequences? Do you know of any software relying on GATB which would have done that?

Thanks!

**Comment by rizkg** [comment]
Hi,

I did that some time ago using htslib, but did not push it to GATB master branch because of the added dependency to htslib that I thought would bother some people. Let me see if I can put that in a branch or fork.

**Comment by rchikhi** [comment]
another way would be a wrapper script with named pipes? e.g. https://github.com/COMBINE-lab/salmon/issues/29#issuecomment-168534703

**Comment by mikael-s** [comment]
> I did that some time ago using htslib, but did not push it to GATB master branch because of the added dependency to htslib that I thought would bother some people. Let me see if I can put that in a branch or fork.

Awesome, thanks!

> another way would be a wrapper script with named pipes? e.g. COMBINE-lab/salmon#29 (comment)

Named pipes are nice but unfortunately I may need to read the input file more than once.

**Comment by rchikhi** [comment]
yes exactly, I came back to post this, sorry this pipe solution can't work in gatb as it already reads the file more than once internally
